ufl/algorithms/optimise_complex_nodes.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+"""Algorithm for removing superfluous complex node combinations."""
+
+from ufl.corealg.multifunction import MultiFunction
+from ufl.constantvalue import Zero
+from ufl.algebra import Abs, Conj, Real, Imag
+from ufl.algorithms.map_integrands import map_integrand_dags
+
+
+class ComplexNodeOptimisation(MultiFunction):
+    """Removes superfluous complex node combinations."""
+    def __init__(self):
+        MultiFunction.__init__(self)
```

Comment:
No need to implement default constructor.

Reply:
:+1: